app/components/sidebars/settings/settings_sidebar_base.js

```diff
@@ -269,23 +272,34 @@ export default class SettingsSidebarBase extends PureComponent {
             </View>
         );
 
-        const clearButton = isStatusSet ?
-            (
-                <ClearButton
-                    handlePress={this.clearCustomStatus}
+        const customStatusExpiryTime = isStatusSet && customStatus?.duration && isCustomStatusExpirySupported ? (
+            <Text style={style.customStatusExpiryContainer}>
+                <CustomStatusExpiry
+                    time={customStatus?.expires_at}
                     theme={theme}
-                    testID='settings.sidebar.custom_status.action.clear'
+                    textStyles={style.customStatusExpiryText}
+                    withinBrackets={true}
+                    showPrefix={true}
+                    testID={'custom_status.expiry'}
                 />
-            ) : null;
-
-        const retryMessage = showRetryMessage ?
-            (
-                <FormattedText
-                    id='custom_status.failure_message'
-                    defaultMessage='Failed to update status. Try again'
-                    style={style.retryMessage}
-                />
-            ) : null;
+            </Text>
+        ) : null;
+
+        const clearButton = isStatusSet ? (
+            <ClearButton
+                handlePress={this.clearCustomStatus}
+                theme={theme}
+                testID='settings.sidebar.custom_status.action.clear'
+            />
+        ) : null;
+
+        const retryMessage = showRetryMessage ? (
+            <FormattedText
+                id='custom_status.failure_message'
+                defaultMessage='Failed to update status. Try again'
+                style={style.retryMessage}
+            />
+        ) : null;
```

Comment:
no need for defining those in variables ... please add those directly into the return value